### PR TITLE
4973 api get monthly volume

### PR DESCRIPTION
**Pull request by NNagato**

### CODE REVIEW

File: tradelogs/http/monthly_volume_handler.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	Reserve string `form:"reserve" binding:"isAddress"`
+}
+
+func (sv *Server) getMonthlyVolume(c *gin.Context) {
+	var (
+		query monthlyVolumeQuery
+	)
+	if err := c.ShouldBindQuery(&query); err != nil {
+		c.JSON(
+			http.StatusBadRequest,
```
Comment (ducquangkstn) [inline tradelogs/http/monthly_volume_handler.go]: using 'httputil.ResponseFailure' for simple code

File: tradelogs/storage/influx/volume_influxdb.go [hunk 2]
```diff
@@ -178,3 +185,42 @@ func convertRowValueToVolume(v []interface{}, idxs volSchema.FieldsRegistrar) (u
...
+			"func", caller.GetCurrentFunctionName(),
+			"reserve", reserveAddr.Hex(),
+			"from", fromTime,
+			"to", toTime,
+		)
+		result        = make(map[uint64]*common.VolumeStats)
+		reserveFilter string
+	)
+
+	if reserveAddr.Hex() != (ethereum.Address{}.Hex()) {
```
Comment (ducquangkstn) [inline tradelogs/storage/influx/volume_influxdb.go]: I would recommend to use func 'blockchain.IsZeroAddress()'

File: tradelogs/storage/influx/volume_influxdb.go [hunk 3]
```diff
@@ -136,7 +137,13 @@ func convertQueryResultToVolume(row influxModel.Row) (map[uint64]*common.VolumeS
 		if err != nil {
 			return nil, err
 		}
-		result[ts] = vol
+		if result[ts] != nil {
```
Comment (ducquangkstn) [inline tradelogs/storage/influx/volume_influxdb.go]: using `preVol, ok = result[ts]` would be recommend

File: tradelogs/http/monthly_volume_handler.go (hunk 1 above)
Comment (NNagato) [reply tradelogs/http/monthly_volume_handler.go]: changed in https://github.com/KyberNetwork/reserve-stats/pull/399/commits/a45a257f397e523354ddc4245ce40fe260ded19b

File: tradelogs/storage/influx/volume_influxdb.go (hunk 2 above)
Comment (NNagato) [reply tradelogs/storage/influx/volume_influxdb.go]: changed in https://github.com/KyberNetwork/reserve-stats/pull/399/commits/a45a257f397e523354ddc4245ce40fe260ded19b

File: tradelogs/storage/influx/volume_influxdb.go (hunk 3 above)
Comment (NNagato) [reply tradelogs/storage/influx/volume_influxdb.go]: changed in https://github.com/KyberNetwork/reserve-stats/pull/399/commits/a45a257f397e523354ddc4245ce40fe260ded19b

**Pull request merged**